### PR TITLE
implemented endpoints access by roles and user authorization

### PR DESCRIPTION
**Pull request by DmytroV95**

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -7,3 +7,6 @@ server.servlet.context-path=/api
 spring.jpa.hibernate.ddl-auto=validate
 spring.jpa.show-sql=true
 spring.jpa.open-in-view=false
+
+jwt.secret=fdkvldhgohagohadogabcdefghijhjhjhfjgfhjghSDD
```
Comment (IhorKalaur) [inline src/main/resources/application.properties:11]: better to use random words for readability

**Review by IhorKalaur — APPROVED** [review]
Good, also you can create changeSet for adding admin to you DB

File: README.md [hunk 2]
```diff
@@ -1 +1,7 @@
-# online-book-store
\ No newline at end of file
+# online-book-store
+
+There is a list of all entities that presented in the project:
```
Comment (andrii-hoienko) [inline README.md:3]: Are these changes really needed?

File: src/main/java/com/varukha/onlinebookstore/security/CustomUserDetailsService.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+@Component
+@RequiredArgsConstructor
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String username)
+            throws UsernameNotFoundException {
+        return userRepository.findUserByEmail(username)
+                .orElseThrow(() -> new RuntimeException("Can't find user by email"));
```
Comment (andrii-hoienko) [inline src/main/java/com/varukha/onlinebookstore/security/CustomUserDetailsService.java:19]: It is bad practice to throw RuntimeException, better to create a custom or use an existing one but more specific

File: src/main/java/com/varukha/onlinebookstore/service/impl/UserServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    private final RoleRepository roleRepository;
+
+    @Override
+    public UserRegistrationResponseDto register(UserRegistrationRequestDto request)
+            throws RegistrationException {
+        if (userRepository.findUserByEmail(request.getEmail()).isPresent()) {
+            throw new RegistrationException("Unable complete registration");
+        }
+        User user = userMapper.toModel(request);
+        user.setPassword(passwordEncoder.encode(request.getPassword()));
```
Comment (andrii-hoienko) [inline src/main/java/com/varukha/onlinebookstore/service/impl/UserServiceImpl.java:35]: I would suggest to make few methods here to separate responsibilities, like: getRole, createUser, maybe some other methods

File: src/main/java/com/varukha/onlinebookstore/validation/EmailValidator.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import jakarta.validation.ConstraintValidatorContext;
+import java.util.regex.Pattern;
+
+public class EmailValidator implements ConstraintValidator<Email, String> {
+    private static final String PATTERN_OF_EMAIL =
+            "^[a-zA-Z0-9.%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$";
+
+    @Override
+    public boolean isValid(String email, ConstraintValidatorContext constraintValidatorContext) {
+        return email != null && !email.isEmpty() && !email.isBlank()
```
Comment (andrii-hoienko) [inline src/main/java/com/varukha/onlinebookstore/validation/EmailValidator.java:13]: Please double-check that there are no duplications, like isEmpty probably checks null case

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 6]
```diff
@@ -3,3 +3,11 @@ databaseChangeLog:
...
   - include:
       file: db/changelog/changes/02-add-is_deleted-column-to-book-table.yaml
+  - include:
+      file: db/changelog/changes/03-create-user-table.yaml
+  - include:
+      file: db/changelog/changes/04-create-role-table.yaml
+  - include:
+      file: db/changelog/changes/05-create-user-role-table.yaml
+  - include:
+      file: db/changelog/changes/06-add-roles-to-role-table.yaml
```
Comment (andrii-hoienko) [inline src/main/resources/db/changelog/db.changelog-master.yaml:13]: Please add empty line in the end of file

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Nice, but some improvements required

File: README.md (hunk 2 above)
Comment (DmytroV95) [reply README.md:3]: Each task have new information and logic to implement, so I try to get information to describe my project.

**Review by vovasalyha — APPROVED** [review]
